adam/curriculum/verbs_with_dynamic_prepositions_curriculum.py

```diff
@@ -0,0 +1,443 @@
+                    (AGENT, agent),
+                    (GOAL, Region(goal_object, distance=PROXIMAL)),
+                ],
+            )
+        ],
+        gazed_objects=[agent],
+    )
+
+
+def _go_in_template(
+    agent: TemplateObjectVariable,
+    goal_object: TemplateObjectVariable,
+    background: Iterable[TemplateObjectVariable],
+) -> Phase1SituationTemplate:
+    return Phase1SituationTemplate(
+        f"go_in-{agent.handle}-in-{goal_object.handle}",
+        salient_object_variables=[agent, goal_object],
+        background_object_variables=background,
+        actions=[
+            Action(
+                GO,
+                argument_roles_to_fillers=[
+                    (AGENT, agent),
+                    (GOAL, Region(goal_object, distance=INTERIOR)),
+                ],
+            )
+        ],
+        constraining_relations=[bigger_than(goal_object, agent)],
+    )
+
+
+def _go_beside_template(
+    agent: TemplateObjectVariable,
+    goal_object: TemplateObjectVariable,
+    background: Iterable[TemplateObjectVariable],
+    *,
+    is_right: bool,
+    is_distal: bool,
+) -> Phase1SituationTemplate:
+    return Phase1SituationTemplate(
+        f"go_beside-{agent.handle}-beside-{goal_object.handle}",
+        salient_object_variables=[agent, goal_object],
+        background_object_variables=background,
+        actions=[
+            Action(
+                GO,
+                argument_roles_to_fillers=[
+                    (AGENT, agent),
+                    (
+                        GOAL,
+                        Region(
+                            goal_object,
+                            distance=DISTAL if is_distal else PROXIMAL,
+                            direction=Direction(
+                                positive=is_right,
+                                relative_to_axis=HorizontalAxisOfObject(
+                                    goal_object, index=0
+                                ),
+                            ),
+                        ),
+                    ),
+                ],
+            )
+        ],
+    )
+
+
+def _go_behind_template(
+    agent: TemplateObjectVariable,
+    goal_object: TemplateObjectVariable,
+    background: Iterable[TemplateObjectVariable],
+    *,
+    is_distal: bool,
+) -> Phase1SituationTemplate:
+    return Phase1SituationTemplate(
+        f"go_behind-{agent.handle}-behind-{goal_object.handle}",
+        salient_object_variables=[agent, goal_object],
+        background_object_variables=background,
+        actions=[
+            Action(
+                GO,
+                argument_roles_to_fillers=[
+                    (AGENT, agent),
+                    (
+                        GOAL,
+                        Region(
+                            goal_object,
+                            distance=DISTAL if is_distal else PROXIMAL,
+                            direction=Direction(
+                                positive=False,
```

Comment:
Can you try to think through what changes we need so that we can account for the positive direction of an axis not being correlated with whether it faces a given person?

We might need to modify `AxisFunctions` to return axis-sign tuples.